src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryResource.java

```diff
@@ -302,4 +315,47 @@ abstract Repository getRepository(Long domainId, RepositoryActionType repository
         }
         return responseEntitySuccess;
     }
+
+    /**
+     * Iterate through the file submissions and try to save each one. Will continue iterating when an error is encountered on updating a file and store it's error in the resulting
+     * Map.
+     *
+     * @param submissions the file submissions (changes) that should be saved in the repository
+     * @param repository the git repository in which the file changes should be saved
+     * @return a map of <filename, error | null>
+     */
+    protected Map<String, String> saveFileSubmissions(List<FileSubmission> submissions, Repository repository) {
+        // If updating the file fails due to an IOException, we send an error message for the specific file and try to update the rest
+        Map<String, String> fileSaveResult = new HashMap<>();
+        submissions.forEach((submission) -> {
+            try {
+                fetchAndUpdateFile(submission, repository);
+                fileSaveResult.put(submission.getFileName(), null);
+            }
+            catch (IOException ex) {
+                fileSaveResult.put(submission.getFileName(), ex.getMessage());
+            }
+        });
+        return fileSaveResult;
+    }
+
+    /**
+     * Retrieve the file from repository and update its content with the submission's content. Throws exceptions if the user doesn't have permissions, the file can't be retrieved
+     * or it can't be updated.
+     *
+     * @param submission information about file update
+     * @param repository repository in which to fetch and update the file
+     * @throws IOException exception when the file in the file submission parameter is empty
+     */
+    private void fetchAndUpdateFile(FileSubmission submission, Repository repository) throws IOException {
+        Optional<File> file = gitService.getFileByName(repository, submission.getFileName());
+
+        if (file.isEmpty()) {
+            throw new IOException("File could not be found.");
+        }
+
+        InputStream inputStream = new ByteArrayInputStream(submission.getFileContent().getBytes(StandardCharsets.UTF_8));
+        Files.copy(inputStream, file.get().toPath(), StandardCopyOption.REPLACE_EXISTING);
+        inputStream.close();
```

Comment:
I would suggest to use
```suggestion
        Files.writeString(file.get().toPath(), submission.getFileContent());
```
unless we have a good reason to use copy here (we should add a comment then).